packages/core/src/lib/lit-core.ts

```diff
@@ -301,7 +301,41 @@ export class LitCore {
       stakingContract.on('StateChanged', async (state: StakingStates) => {
         log(`New state detected: "${state}"`);
         if (state === StakingStates.NextValidatorSetLocked) {
+          log(
+            'State found to be new validator set locked, checking validator set'
+          );
+          let oldNodeUrls: string[] = [...this.config.bootstrapUrls].sort();
           await this.setNewConfig();
+          let currentNodeUrls: string[] = this.config.bootstrapUrls.sort();
+          let delta: string[] = currentNodeUrls.filter(
+            (item) => oldNodeUrls.indexOf(item) > -1
+          );
+          // if the sets differ we reconnect.
+          if (delta.length > 1) {
+            // check if the node sets are non matching and re connect if they do not.
+            /*
+              TODO: While this covers most cases where a node may come in or out of the active 
+              set which we will need to re attest to the execution enviorments.
+              The sdk currently does not know if there is an active network operation pending.
+              Such that the state when the request was sent will now mutate when the response is sent back.
+              The sdk should be able to understand its current execution enviorment and wait on an active 
+              network request to the previous epoch's node set before changing over.
+              
+            */
+            log(
+              'Active validator sets changed, new validators ',
+              delta,
+              'starting node connection'
+            );
+            await this._runHandshakeWithBootstrapUrls().catch(
```

Comment:
Before this we should clear `this.connectedNodes` because `_runHandshakeWithBootstrapUrls` will try to re-add every handshaked node
The old ones will stay, the new ones will be added and the ones that didn't change will stay with the same value (there is no update on `Set.add`)

Reply:
Good call, yes quite right. I have been used to using `indexOf` for a long time. switch to `includes` but kept the top level `filter`.